diff --git a/super/CMakeLists.txt b/super/CMakeLists.txt
index 7f76221ec..c22d4fb71 100644
--- a/super/CMakeLists.txt
+++ b/super/CMakeLists.txt
@@ -52,6 +52,8 @@ ExternalProject_Add(
         LIST_SEPARATOR
         |
     CMAKE_ARGS -G${CMAKE_GENERATOR}
+               -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+               -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
                -DCMAKE_PREFIX_PATH=${GOOGLE_CLOUD_CPP_PREFIX_PATH}
                -DCMAKE_INSTALL_RPATH=${GOOGLE_CLOUD_CPP_INSTALL_RPATH}
                -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
diff --git a/super/external/google-cloud-cpp-common.cmake b/super/external/google-cloud-cpp-common.cmake
index b979af710..84558d3fd 100644
--- a/super/external/google-cloud-cpp-common.cmake
+++ b/super/external/google-cloud-cpp-common.cmake
@@ -48,6 +48,7 @@ if (NOT TARGET google-cloud-cpp-common-project)
                    -DCMAKE_PREFIX_PATH=${GOOGLE_CLOUD_CPP_PREFIX_PATH}
                    -DCMAKE_INSTALL_RPATH=${GOOGLE_CLOUD_CPP_INSTALL_RPATH}
                    -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+		   -DGOOGLE_CLOUD_CPP_ENABLE_MACOS_OPENSSL_CHECK=OFF
         BUILD_COMMAND ${CMAKE_COMMAND} --build <BINARY_DIR> ${PARALLEL}
         LOG_DOWNLOAD ON
         LOG_CONFIGURE OFF
